<a href="https://colab.research.google.com/github/HenriqueCCdA/bootCampAluraDataScience/blob/master/modulo6/desafios/modelo6_aula1_desafio.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [2]:
df_clientes_cadastrados = pd.read_csv('https://raw.githubusercontent.com/HenriqueCCdA/bootCampAluraDataScience/master/modulo6/Dados/clientes_cadastrados.csv')
df_clientes_cadastrados.head()

ID_Cliente Genero  Tem_Carro  ...   Ocupacao  Tamanho_Familia  Anos_empregado
0     5008804      H          1  ...     Outros              2.0            12.4
1     5008805      H          1  ...     Outros              2.0            12.4
2     5008806      H          1  ...  Seguranca              2.0             3.1
3     5008808      M          0  ...      Venda              1.0             8.4
4     5008809      M          0  ...      Venda              1.0             8.4

[5 rows x 18 columns]

# Desafio 1: Explorar IDs repetidos (Atualização de cadastro? Erro?).

In [14]:
id_repetidos = df_clientes_cadastrados[df_clientes_cadastrados['ID_Cliente'].duplicated(keep=False)]['ID_Cliente'].unique()

array(['H', 'M'], dtype=object)

In [22]:
for id in id_repetidos[:20]: 
  display(df_clientes_cadastrados.query('ID_Cliente == @id'))

ID_Cliente Genero  Tem_Carro  ...  Ocupacao  Tamanho_Familia  Anos_empregado
421211     7702516      M          0  ...     Venda              4.0             3.4
432379     7702516      M          0  ...  Medicina              3.0            11.7

[2 rows x 18 columns]

ID_Cliente Genero  ...  Tamanho_Familia  Anos_empregado
421268     7602432      H  ...              2.0             3.6
421349     7602432      M  ...              2.0         -1000.7

[2 rows x 18 columns]

ID_Cliente Genero  Tem_Carro  ...  Ocupacao  Tamanho_Familia  Anos_empregado
421464     7836971      H          1  ...    Outros              3.0            15.1
428620     7836971      M          0  ...     Venda              2.0             7.7

[2 rows x 18 columns]

ID_Cliente Genero  ...  Tamanho_Familia  Anos_empregado
421698     7213374      H  ...              2.0             2.6
425724     7213374      H  ...              3.0             2.9

[2 rows x 18 columns]

ID_Cliente Genero  ...  Tamanho_Familia  Anos_empregado
421726     7052783      H  ...              2.0             7.1
422660     7052783      H  ...              4.0             7.4

[2 rows x 18 columns]

ID_Cliente Genero  ...  Tamanho_Familia  Anos_empregado
421907     7023651      H  ...              3.0             4.0
425306     7023651      M  ...              1.0             3.3

[2 rows x 18 columns]

ID_Cliente Genero  ...  Tamanho_Familia  Anos_empregado
422068     7838075      H  ...              2.0             3.5
423702     7838075      M  ...              1.0             4.5

[2 rows x 18 columns]

ID_Cliente Genero  ...  Tamanho_Familia  Anos_empregado
422077     7636389      H  ...              2.0             1.8
427167     7636389      M  ...              2.0             1.4

[2 rows x 18 columns]

ID_Cliente Genero  ...  Tamanho_Familia  Anos_empregado
422119     7052812      M  ...              1.0             0.3
431414     7052812      M  ...              2.0            12.4

[2 rows x 18 columns]

ID_Cliente Genero  ...  Tamanho_Familia  Anos_empregado
422326     7372589      M  ...              1.0         -1000.7
432562     7372589      M  ...              1.0             3.0

[2 rows x 18 columns]

ID_Cliente Genero  ...  Tamanho_Familia  Anos_empregado
422747     7155150      M  ...              2.0             3.9
429029     7155150      M  ...              4.0             5.7

[2 rows x 18 columns]

ID_Cliente Genero  ...  Tamanho_Familia  Anos_empregado
423076     7090931      M  ...              2.0            14.6
429834     7090931      M  ...              2.0             4.1

[2 rows x 18 columns]

ID_Cliente Genero  Tem_Carro  ...  Ocupacao  Tamanho_Familia  Anos_empregado
423143     7091721      H          1  ...    Outros              2.0            24.3
433666     7091721      M          1  ...  Medicina              2.0             6.2

[2 rows x 18 columns]

ID_Cliente Genero  ...  Tamanho_Familia  Anos_empregado
423160     7137299      H  ...              3.0            19.9
426665     7137299      M  ...              1.0             5.7

[2 rows x 18 columns]

ID_Cliente Genero  Tem_Carro  ...  Ocupacao  Tamanho_Familia  Anos_empregado
423327     7744386      M          1  ...   Cozinha              2.0             1.1
431391     7744386      H          1  ...  Gerencia              1.0             1.4

[2 rows x 18 columns]

ID_Cliente Genero  Tem_Carro  ...  Ocupacao  Tamanho_Familia  Anos_empregado
423416     7772847      M          1  ...     Venda              2.0             4.5
429104     7772847      M          0  ...    Outros              1.0         -1000.7

[2 rows x 18 columns]

ID_Cliente Genero  ...  Tamanho_Familia  Anos_empregado
423592     7636756      M  ...              3.0             1.2
432355     7636756      M  ...              3.0             0.8

[2 rows x 18 columns]

ID_Cliente Genero  ...  Tamanho_Familia  Anos_empregado
423794     7317997      H  ...              4.0             2.6
429053     7317997      M  ...              4.0             0.4

[2 rows x 18 columns]

ID_Cliente Genero  ...  Tamanho_Familia  Anos_empregado
423852     7053557      H  ...              3.0             0.7
424480     7053557      H  ...              3.0             6.2

[2 rows x 18 columns]

ID_Cliente Genero  ...  Tamanho_Familia  Anos_empregado
423874     7742853      M  ...              3.0             1.9
433159     7742853      H  ...              1.0             4.6

[2 rows x 18 columns]

* A maioria dos ids repetidos parecem clientes diferentes, um possivel erro no procedimento de casdatramento.
* Alguns ids parecem com o valor Anos_empregado como -1000.7 para uma das entradas. O que poderia indicar uma erro apenas dessa colunas. Porém a coluna idade pare completamente diferente 

> O **ID_CLiente** geralmente é um campo muito importante de banco de dados. Se há um erro nesse campo eu tenho bastantece receio na confiabilidade dos outro campos. Por isso acho que a melhor opção é a exlusão tatal dessa entradas da base de dados.

# Desafio 2: Explorar as colunas (features) identificar continuas, categóricas e binárias.

In [23]:
df_clientes_cadastrados.head(n=2)

ID_Cliente Genero  Tem_Carro  ...  Ocupacao  Tamanho_Familia  Anos_empregado
0     5008804      H          1  ...    Outros              2.0            12.4
1     5008805      H          1  ...    Outros              2.0            12.4

[2 rows x 18 columns]

In [ ]:
df_clientes_cadastrados['Genero'].unique()

In [24]:
df_clientes_cadastrados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 438557 entries, 0 to 438556
Data columns (total 18 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   ID_Cliente             438557 non-null  int64  
 1   Genero                 438557 non-null  object 
 2   Tem_Carro              438557 non-null  int64  
 3   Tem_Casa_Propria       438557 non-null  int64  
 4   Qtd_Filhos             438557 non-null  int64  
 5   Rendimento_Anual       438557 non-null  float64
 6   Categoria_de_renda     438557 non-null  object 
 7   Grau_Escolaridade      438557 non-null  object 
 8   Estado_Civil           438557 non-null  object 
 9   Moradia                438557 non-null  object 
 10  Idade                  438557 non-null  float64
 11  Tem_Celular            438557 non-null  int64  
 12  Tem_telefone_trabalho  438557 non-null  int64  
 13  Tem_telefone_fixo      438557 non-null  int64  
 14  Tem_email              438557 non-nu

1. ID_Cliente - Apesar ser **int** essa coluna não se enquandra 
2. Genero - Categorica binaria (Homem, mulher). (OBS: Esta categórica deferia termas opções se tornando uma variveis categórica)
3. Tem_Carro  - Categórica binaria
4. Tem_Casa_Propria - Categórica binária  
5. Qtd_Filhos - Numérica discreta ( apenas numeros interios)
6. Categoria_de_renda - Categórica 